csrc/multidevice/executor.h

```diff
@@ -123,7 +136,7 @@ class MultiDeviceExecutor {
   std::unordered_map<Val*, c10::IValue> val_to_IValue_;
 
   // holds the Communicator to be used for execution
-  Communicator& comm_;
+  Communicator* comm_ = nullptr;
```

Comment:
I'm wondering if this should have been a `std::shared_ptr<Communicator>` instead.